integration-scripts/src/initiateServiceAgreement.ts

```diff
@@ -1,15 +1,18 @@
+import { generated as chainlink } from 'chainlinkv0.5'
+import { contract } from 'chainlink'
 import { helpers } from 'chainlink'
 import {
   getArgs,
   registerPromiseHandler,
   DEVNET_ADDRESS,
   createProvider,
 } from './common'
-import { CoordinatorFactory } from './generated/CoordinatorFactory'
 import { ethers } from 'ethers'
-import { Coordinator } from './generated/Coordinator'
+const { CoordinatorFactory } = chainlink
 
-type CoordinatorParams = Parameters<Coordinator['initiateServiceAgreement']>
+type CoordinatorParams = Parameters<
+  contract.Instance<chainlink.CoordinatorFactory>['initiateServiceAgreement']
```

Comment:
Maybe there's no need for the `chainlink.` prefix, since `CoordinatorFactory` was brought into scope on line 11.

Reply:
Unfortunately we need the type here, not the value. Typescript differentiates between the two. Line 11 gives me the value but not the type, and `typeof` gives me the constructor value rather than the class type